pages/emailSender.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { useState } from 'react'
+import {
+  ThemeProvider,
+  Divider,
+  Typography,
+  FormControl,
+  TableContainer,
+  Paper,
+  TableBody,
+  TableCell,
+  TableHead,
+  Button
+} from '@mui/material'
+import type { NextPage } from 'next'
+import { nanoid } from 'nanoid'
+import { useTheme } from '@mui/material/styles'
+import { StyledButton, StyledPageContainer } from '../styles/common'
+import { CsvRow, ReplaceObj, Message } from '../lib/types'
+import {
+  SectionContainer,
+  StyledCsvButton,
+  StyledTextArea,
+  StyledCsvButtonsContainer,
+  StyledSubHeader,
+  StyledFinalMessagesContainer,
+  StyledTableContainer,
+  StyledDivider,
+  StyledTable,
+  StyledTableRow
+} from '../pageStyles/emailSender.styles'
+
+const EmailSender: NextPage = () => {
+  const [file, setFile] = useState()
+  const [csvRowsArray, setCsvRowsArray] = useState<CsvRow[]>([])
+  const [message, setMessage] = useState('')
+  const [finalMessages, setFinalMessages] = useState<Message[]>([])
+  const theme = useTheme()
+
+  let reader: FileReader
+  if (typeof window !== 'undefined') {
+    reader = new window.FileReader()
+  }
+  const handleOnChange = (e: any) => {
+    setFile(e.target.files[0])
+  }
+
+  const csvFileToArray = (str: string) => {
+    const csvHeader = str.slice(0, str.indexOf('\n')).split(',')
+    const csvRows = str.slice(str.indexOf('\n') + 1).split('\n')
+    const array = csvRows.map((i) => {
```

Comment:
we can name the variables a bit better here - "array", "obj", etc. Had some trouble understanding what was going on

Reply:
LOL agreed